src/MediatR/Mediator.cs

```diff
@@ -226,6 +256,18 @@ private IEnumerable<TWrapper> GetNotificationHandlers<TWrapper>(object notificat
                 .ToList();
         }
 
+        private Task<object> CreatePipeline(object request, RequestHandlerDelegate invokeHandler)
+        {
+            var behaviors = _multiInstanceFactory(typeof(IPipelineBehavior))

```

Comment:
Yeah, I would put it in the pipeline. Although if I made the pipeline generic, then the container could pick the correct instances.